### PR TITLE
Add --insecure flag to podman manifest inspect for Docker compatibility

### PR DESCRIPTION
**Pull request by rhatdan**
Helps fix: https://github.com/containers/podman/issues/14917

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
podman manifest inspect now supports --insecure flag for Docker compatibility
```

### CODE REVIEW

File: test/system/012-manifest.bats
Comment (edsantiago) [inline test/system/012-manifest.bats:14]: Aaaaaaaahhhhhh! Evil! I must never have reviewed the initial commit. Please fix this to
```bash
    assert "$output" =~ "sha256:$iid" "Original image ID still shown in podman-images output"
```

File: test/system/012-manifest.bats
Comment (edsantiago) [inline test/system/012-manifest.bats:14]: Nice addition, but comment ("Original image ID...") is misleading. Probably not worth the CI cost to re-push, but if you do, could you also add a comment something like "# --insecure is a NOP, we use it here just to make sure podman accepts it"

File: cmd/podman/manifest/inspect.go
Comment (mheon) [inline cmd/podman/manifest/inspect.go:30]: This seems like it should force `--tls-verify=force` instead of doing nothing

Comment (rhatdan) [reply cmd/podman/manifest/inspect.go:30]: tls-verify is not supported yet for podma manifest inspect, that I can see.

Comment (rhatdan) [reply cmd/podman/manifest/inspect.go:30]: I am not sure it even talks to a remote service.

@nalind WDYT?

Comment (nalind) [reply cmd/podman/manifest/inspect.go:30]: It connects to a registry if there's no local list with the specified name, I think since #7735.

Comment (vrothberg) [reply cmd/podman/manifest/inspect.go:30]: Yes, that sounds good.  `--insecure` can then also just alias to `--tls-verify`.

Comment (vrothberg) [reply cmd/podman/manifest/inspect.go:30]: Requires passing the value down to `remoteManifestInspect()` and flip the switch in the system context.

File: pkg/domain/infra/tunnel/manifest.go
Comment (nalind) [inline pkg/domain/infra/tunnel/manifest.go:36]: Why is this reusing a different method's options structure instead of rolling its own, as most of the others in this file that take options structures do?

Comment (rhatdan) [reply pkg/domain/infra/tunnel/manifest.go:36]: lazyness?

File: test/system/012-manifest.bats
Comment (nalind) [inline test/system/012-manifest.bats:13]: `$output` looks like it'll be the local manifest's ID, which I wouldn't expect us to ask a registry about.  Is this testing the right thing?

Comment (nalind) [reply test/system/012-manifest.bats:13]: The `podman login` tests fire up a registry container with TLS configured.  If we did that here, so long as the client isn't configured to trust the CA certificate (HTTPS, but not verifiable), or there is no certificate (plain old HTTP), a client will have to have a working `--insecure` flag in order to be able to talk to it at all.  But we'd have to be looking for an image using a name rather than an ID.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by rhatdan** [comment]
@vrothberg @mheon @edsantiago PTAL

**Review by edsantiago — CHANGES_REQUESTED** [review]
Ugh, looks like I need to write a new CI check. For now, can you please:
* `git rm docs/source/markdown/podman-manifest-instpect.1.md`
* manually add `podman-manifest-inspect.1.md` to the file `docs/source/markdown/.gitignore` (preferably in sorted order)
* re-push

Otherwise LGTM, although all I reviewed was docs & tests.

**Review by edsantiago — APPROVED** [review]

**Comment by edsantiago** [comment]
CI is finally green (lots o' flakes). LGTM but this needs more approval than just mine.

**Review by vrothberg — APPROVED** [review]
/lgtm

**Pull request merged**
